src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -4,28 +4,109 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import importlib
+import json
 import logging
 from abc import ABC, abstractproperty
 from collections import OrderedDict
+from enum import Enum
+from pathlib import Path
 from typing import TYPE_CHECKING, Any
 
 import lightning.pytorch as pl
+import numpy as np
 import torch
 from lightning.pytorch.trainer.states import TrainerFn
 from lightning.pytorch.utilities.types import STEP_OUTPUT
 from torch import nn
-from torchvision.transforms.v2 import Compose, Normalize, Resize, Transform
+from torchvision.transforms.v2 import CenterCrop, Compose, Normalize, Resize, Transform
 
-from anomalib import LearningType
+from anomalib import LearningType, TaskType
+from anomalib.data.transforms import ExportableCenterCrop
 from anomalib.metrics import AnomalibMetricCollection
 from anomalib.metrics.threshold import BaseThreshold
+from anomalib.utils.exceptions import try_import
 
 if TYPE_CHECKING:
     from lightning.pytorch.callbacks import Callback
+    from torch.types import Number
     from torchmetrics import Metric
 
 
 logger = logging.getLogger(__name__)
+if try_import("openvino"):
+    from openvino.runtime import serialize
+    from openvino.tools.ovc import convert_model
+
+
+class ExportType(str, Enum):
+    """Model export type.
+
+    Examples:
+        >>> from anomalib.models import ExportType
+        >>> ExportType.ONNX
+        'onnx'
+        >>> ExportType.OPENVINO
+        'openvino'
+        >>> ExportType.TORCH
+        'torch'
+    """
+
+    ONNX = "onnx"
+    OPENVINO = "openvino"
+    TORCH = "torch"
+
+
+class InferenceModel(nn.Module):
```

Comment:
Also not sure if this is the best place for `ExportType`

Reply:
Maybe we can move it to `src/uitls/types`

Reply:
I agree that `InferenceModel` should be in a separate file, preferably under `anomalib.deploy`.

Concerning `ExportType`, I think we previously agreed to place enums in the root `__init__` if central to the library's functionality, or otherwise keep them within the submodule. In this case, `ExportType` is only used by the engine and users would likely not have access it (they can just call the new export entrypoints of the model directly), so I feel it should be contained in the `anomalib.deploy` submodule.

Reply:
I simply moved `InferenceModel` and `ExportType` back to `anomalib.deploy.export`. This way, we minimize the changes to the codebase.